### PR TITLE
Fix Exception in a_generate_reply when function_call is None

### PR DESCRIPTION
**Pull request by artero**
## Why are these changes needed?

After executing `generate_reply` or `a_generate_reply` for an agent with a tool, the generated response (using the GPT-4o-mini model) contains `"function_call"=None`. If we call that function again, it causes an exception in the case of a_generate_reply (but not in generate_reply).

## Related issue number

None

## Checks

- [x] I've included any doc changes needed for https://microsoft.github.io/autogen/. See https://microsoft.github.io/autogen/docs/Contribute#documentation to build and test documentation locally.
- [x] I've added tests (if relevant) corresponding to the changes introduced in this PR.
- [x] I've made sure all auto checks have passed.

### CODE REVIEW

**Comment by artero** [comment]
@microsoft-github-policy-service agree

**Comment by umermansoor** [comment]
lgtm

File: autogen/agentchat/conversable_agent.py [hunk 1]
```diff
@@ -1581,7 +1581,7 @@ async def a_generate_function_call_reply(
         if messages is None:
             messages = self._oai_messages[sender]
         message = messages[-1]
-        if "function_call" in message:
+        if "function_call" in message and message["function_call"]:
             func_call = message["function_call"]
```
Comment (sonichi) [inline autogen/agentchat/conversable_agent.py:1585]:
```suggestion
        func_call = message.get("function_call")
        if func_call:
```

**Review by sonichi — COMMENTED** [review]
Thanks. I made a small suggestion. Could you create the PR from a branch in the upstream repo so that we can run the openai test?

File: autogen/agentchat/conversable_agent.py (hunk 1 above)
Comment (WaelKarkoub) [reply autogen/agentchat/conversable_agent.py:1585]: @artero with this change the PR is merge-able

Comment (artero) [reply autogen/agentchat/conversable_agent.py:1585]: Hi @sonichi, I was on vacation and the invitation sent to me has expired. Could you please resend it?

Comment (thinkall) [reply autogen/agentchat/conversable_agent.py:1585]: @ekzhu , could you help here? Thanks.

Comment (artero) [reply autogen/agentchat/conversable_agent.py]: I incorporated the suggestion and rebased my branch with the current main branch.​

**Review by jackgerrits — APPROVED** [review]

**Pull request merged**